server/databases_test.go

```diff
@@ -308,6 +315,248 @@ func TestService_DropRetentionPolicy(t *testing.T) {
+		{
+			name: "Fails when invalid offset value provided",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, srcID int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID: 0,
+						}, nil
+					},
+				},
+			},
+			args: args{
+				queryParams: map[string]string{
+					"offset": "bob",
+				},
+			},
+			wants: wants{
+				statusCode: 422,
+				body:       `{"code":422,"message":"strconv.Atoi: parsing \"bob\": invalid syntax"}`,
+			},
+		},
+		{
+			name: "Overrides limit less than or equal to 0 with limit 100",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, srcID int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID: 0,
+						}, nil
+					},
+				},
+				Databases: &mocks.Databases{
+					ConnectF: func(context.Context, *chronograf.Source) error {
+						return nil
+					},
+					GetMeasurementsF: func(ctx context.Context, dbID string, limit, offset int) ([]chronograf.Measurement, error) {
+						return []chronograf.Measurement{
+							{
+								Name: "pineapple",
+							},
+							{
+								Name: "cubeapple",
+							},
+							{
+								Name: "pinecube",
+							},
+						}, nil
+					},
+				},
+			},
+			args: args{
+				queryParams: map[string]string{
+					"limit": "0",
+				},
+			},
+			wants: wants{
+				statusCode: 200,
+				body: `{"measurements":[{"name":"pineapple"},{"name":"cubeapple"},{"name":"pinecube"}],"links":{"self":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","first":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","next":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=100"}}
+`,
+			},
+		},
+		{
+			name: "Overrides offset less than 0 with offset 0",
+			fields: fields{
+				SourcesStore: &mocks.SourcesStore{
+					GetF: func(ctx context.Context, srcID int) (chronograf.Source, error) {
+						return chronograf.Source{
+							ID: 0,
+						}, nil
+					},
+				},
+				Databases: &mocks.Databases{
+					ConnectF: func(context.Context, *chronograf.Source) error {
+						return nil
+					},
+					GetMeasurementsF: func(ctx context.Context, dbID string, limit, offset int) ([]chronograf.Measurement, error) {
+						return []chronograf.Measurement{
+							{
+								Name: "pineapple",
+							},
+							{
+								Name: "cubeapple",
+							},
+							{
+								Name: "pinecube",
+							},
+						}, nil
+					},
+				},
+			},
+			args: args{
+				queryParams: map[string]string{
+					"offset": "-1337",
+				},
+			},
+			wants: wants{
+				statusCode: 200,
+				body: `{"measurements":[{"name":"pineapple"},{"name":"cubeapple"},{"name":"pinecube"}],"links":{"self":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","first":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=0","next":"/chronograf/v1/sources/0/dbs/0/measurements?limit=100\u0026offset=100"}}
+`,
+			},
+		},
+	}
```

Comment:
We should add a test for
"When requesting limit 100 and offset 200, server includes a `prev` link in the response"

Reply:
good call --- done